code/workspaces/web-app/src/components/settings/UserPermissionsTable.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import Checkbox from '@material-ui/core/Checkbox';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableCell from '@material-ui/core/TableCell';
+import TableBody from '@material-ui/core/TableBody';
+import TableRow from '@material-ui/core/TableRow';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import projectSettingsActions from '../../actions/projectSettingsActions';
+
+const styles = theme => ({
+  activeSelection: {
+    color: theme.palette.primary.main,
+  },
+  implicitSelection: {
+    color: theme.palette.primaryTransparent,
+  },
+  tableHeader: {
+    fontWeight: 'bold',
+    align: 'center',
+  },
+  tableCell: {
+    '&:first-child': {
+      paddingLeft: 0,
+    },
+    '&:last-child': {
+      paddingRight: 0,
+    },
+  },
+});
+
+export const PERMISSIONS = {
+  ADMIN: {
+    name: 'ADMIN',
+    value: 10,
+  },
+  USER: {
+    name: 'USER',
+    value: 5,
+  },
+  VIEWER: {
+    name: 'VIEWER',
+    value: 1,
+  },
+};
+
+export const columnHeadings = [
+  { heading: 'User Name', checkBoxCol: false },
+  { heading: 'Admin', checkBoxCol: true },
+  { heading: 'User', checkBoxCol: true },
+  { heading: 'Viewer', checkBoxCol: true },
+];
+
+const checkBoxColumnOrder = [PERMISSIONS.ADMIN, PERMISSIONS.USER, PERMISSIONS.VIEWER];
+
+export const projectUsersSelector = ({ projectUsers }) => projectUsers;
+
+function UserPermissionsTable({ classes }) {
+  const users = useSelector(projectUsersSelector);
+  const dispatch = useDispatch();
+
+  useEffect(
+    () => {
+      dispatch(projectSettingsActions.getProjectUserPermissions('project'));
+    },
+    [dispatch],
+  );
+
+  return (
+    getTable(users, classes, columnHeadings)
+  );
+}
+
+export function getTable(users, classes, colHeadings) {
+  return (
+    <Table size="small">
+      <TableHead>
+        {getTableHead(columnHeadings, classes)}
+      </TableHead>
+      <TableBody>
+        {getTableBody(users, classes, colHeadings.length)}
+      </TableBody>
+    </Table>
+  );
+}
+
+export function getTableHead(headings, classes) {
+  return (
+    <TableRow>
+      {headings.map((item, index) => (
+        <TableCell
+          className={classes.tableCell}
+          padding={item.checkBoxCol ? 'checkbox' : null}
+          align={item.checkBoxCol ? 'center' : 'left'}
+          key={`header-${index}`}
+        >
+          <Typography className={classes.tableHeader} variant="body1">{item.heading}</Typography>
+        </TableCell>
+      ))}
+    </TableRow>
+  );
+}
+
+export function getTableBody(users, classes, numCols) {
+  if (users.error || !users.value) {
+    return getFullWidthRow(
```

Comment:
Could you make a wrapper of `getFullWidthRow` of `getFullWidthRowForText`? That way you wouldn't need to include the `Typography` every time.